src/libs/blueprint/conduit_blueprint_mesh.cpp

```diff
@@ -2960,6 +2960,260 @@ mesh::topology::unstructured::generate_sides(const Node &topo,
+                }
+            }
+            else
+            {
+                // if we couldn't find the field in the specified field_names, then we don't care
+                // but if it was found, and we are here, then that means that the field we want
+                // uses the wrong topology
+                if (found)
+                {
+                    CONDUIT_ERROR("field " + field_name + " does not use " + topo + ".");
+                }
+            }
+        }
+
+        Node &original_elements = fields_dest["original_element_ids"];
+        original_elements["topology"] = topo;
+        original_elements["association"] = "element";
+        original_elements["volume_dependent"] = "false";
+
+        d2smap["values"].to_uint32_array(original_elements["values"]);
+    }
+}
+
+void
+mesh::topology::unstructured::generate_sides(const conduit::Node &topo_src,
+                                             conduit::Node &topo_dest,
+                                             conduit::Node &coordset_dest,
+                                             conduit::Node &fields_dest,
+                                             conduit::Node &s2dmap,
+                                             conduit::Node &d2smap,
+                                             const conduit::Node &options)
+{
+    std::string field_prefix = "";
+    std::vector<std::string> field_names;
+    const Node &fields_src = (*(topo_src.parent()->parent()))["fields"];
+
+    // check for existence of field prefix
+    if (options.has_child("field_prefix"))
+    {
+        if (options["field_prefix"].dtype().is_string())
+        {
+            field_prefix = options["field_prefix"].as_string();
+        }
+        else
+        {
+            CONDUIT_ERROR("field_prefix must be a string.");
+        }
+    }
+
+    // check for target field names
+    if (options.has_child("field_names"))
+    {
+        if (options["field_names"].dtype().is_string())
+        {
+            field_names.push_back(options["field_names"].as_string());
+        }
+        else if (options["field_names"].dtype().is_list())
+        {
+            NodeConstIterator itr = options["field_names"].children();
+            while (itr.has_next())
+            {
+                const Node &cld = itr.next();
+                if (cld.dtype().is_string())
+                {
+                    field_names.push_back(cld.as_string());
+                }
+                else
+                {
+                    CONDUIT_ERROR("field_names must be a string or a list of strings.");
+                }
+            }
+        }
+        else
+        {
+            CONDUIT_ERROR("field_names must be a string or a list of strings.");
+        }
+    }
+
+    // check that the discovered field names exist in the target fields
+    for (uint64 i = 0; i < field_names.size(); i ++)
+    {
+        bool found = false;
+        NodeConstIterator itr = fields_src.children();
```

Comment:
you can simplify with `fields_src.has_child(field_names[i])`, then you don't need to write your own loop